crates/derive/src/combine_api.rs

```diff
@@ -177,9 +213,24 @@ impl cgo::Service for API {
                     arena,
                     out,
                     &mut state.stats.out,
+                    &mut state.shape,
+                    &mut state.shape_changed,
                 )?;
 
-                if !more {
+                if let Some(ref shape) = state.shape {
```

Comment:
this logic breaks combiner draining, I believe. Did tests not catch it, or just haven't been run yet?

The outer test needs to be `if !more`, which maps into an Accumulator, and otherwise is a Drainer.

schema inference is secondary.

Reply:
Whoops! Tests did catch it, but they hadn't been run until recently